apps/cli/src/credentials-storage.ts

```diff
@@ -7,71 +7,100 @@ export enum CredentialsStorageType {
   ACCOUNT = "user-account",
 }
 
-const keytarShim = {
-  default: {
-    setPassword: async (service: string, account: string, password: string) =>
-      writeFile(join(codemodDirectoryPath, `${service}:${account}`), password),
-    findCredentials: async (service: string) => {
-      const entries = await readdir(codemodDirectoryPath).then((dir) =>
-        dir.filter((file) => file.startsWith(`${service}:`)),
-      );
-
-      return Promise.all(
-        entries.map(async (file) => ({
-          account: file.split(":")[1],
-          password: await readFile(join(codemodDirectoryPath, file), {
-            encoding: "utf-8",
-          }),
-        })),
-      );
-    },
-    deletePassword: async (service: string, account: string) =>
-      unlink(join(codemodDirectoryPath, `${service}:${account}`))
-        .then(() => true)
-        .catch(() => false),
-  },
+const keytar = async () => {
+  return (await import("keytar")).default;
 };
 
 //Sometimes we running production CLI from local build, credentials should be stored in different places.
 const SERVICE = `codemod.com${process.env.NODE_ENV === "production" ? "" : `-${process.env.NODE_ENV}`}`;
 let alreadyWarned = false;
 
-const getKeytar = async () => {
-  try {
-    return await import("keytar");
-  } catch (err) {
-    const isShimLoggedIn = await keytarShim.default
-      .findCredentials(SERVICE)
-      .then((creds) => creds.length > 0);
-
-    if (!alreadyWarned && !isShimLoggedIn) {
-      alreadyWarned = true;
-      console.warn(
-        chalk.red(
-          String(err),
-          `\n\nCodemod CLI uses "keytar" to store your credentials securely.`,
-          `\nPlease make sure you have "libsecret" installed on your system.`,
-          "\nDepending on your distribution, you will need to run the following command",
-          "\nDebian/Ubuntu:",
-          chalk.bold("sudo apt-get install libsecret-1-dev"),
-          "\nFedora:",
-          chalk.bold("sudo dnf install libsecret"),
-          "\nArch Linux:",
-          chalk.bold("sudo pacman -S libsecret"),
-          chalk.cyan(
-            "\n\nIf you were not able to install the necessary package or CLI was not able to detect the installation" +
-              "please reach out to us at our Community Slack channel.",
-          ),
-          chalk.yellow(
-            "\nYou can still use the CLI with file-based replacement that will store your credentials at your home directory.",
-          ),
+async function prepareShim() {
+  await mkdir(codemodDirectoryPath, { recursive: true });
```

Comment:
@arybitskiy I think this PR, and this line specifically, is a great way to handle issue #1397 and should largely supersede my contribution in #1398

I do think you should consider adding a unit test, along the lines of what I wrote in #1398 to confirm the fix works and to prevent future regressions. If you would be open to it, I am happy to split the unit test off and put up a PR against this branch to save you the trouble. #1398 could then be closed.

Reply:
Great idea and contribution! You've already put in the effort, and I'd love to see your name among the PR contributors here. Let's make it happen! thanks so much, @spirulence! 🍻